pkg/protocol/engine/engine.go

```diff
@@ -273,23 +275,9 @@ func (e *Engine) Block(id iotago.BlockID) (*model.Block, bool) {
 	return modelBlock, modelBlock != nil
 }
 
+// IsBootstrapped returns true if the node is bootstrapped. Do not get bootstrapped status with this method, but always get it from SyncManager. This method is needed and exported to access component inside the engine, so we could test it with different bootstrapped conditions.
```

Comment:
It's a little confusing to have a method and discourage its use. I don't see this method used anywhere in tests - are we certain that it needs to be exposed? Or that it needs to exist at all? Maybe we can pass the `optsIsBootstrappedFunc` directly to all the components that need it, instead of `e.IsBootstrapped`?